### ISSUE TITLE
memory leak in DerivedProperty

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/axon#37]
DerivedProperty.detach does this:

``` js
detach: function() {
  for ( var i = 0; i < this.dependencies.length; i++ ) {
      var dependency = this.dependencies[i];
      dependency.unlink( this.dependencyListeners[i] );
  }
}
```

But it neglects to clear the arrays this.dependencies and this.dependencyListeners. Both of these contain references to the derived properties (through a function closure in dependencyListeners). So while the observers are unlinked from the derived properties, the derived property still holds references to the properties that it's derived from.

Here's an example:

``` js
var Property = require( 'AXON/Property' );
var DerivedProperty = require( 'AXON/DerivedProperty' );
var widthProperty = new Property( 2 );
var heightProperty = new Property( 3 );
var areaProperty = new DerivedProperty( [ widthProperty, heightProperty ], 
  function( width, height ) { return width * height; } );
areaProperty.link( function( area ) { console.log( 'area = ' + area );  } );
debugger;
```

In a browser console:

``` js
> widthProperty._observers.length
1 // ok
> heightProperty._observers.length
1 // ok
> areaProperty.dependencies.length
2 // ok
> areaProperty.dependencyListeners.length
2 // ok
> areaProperty.detach()
undefined
> widthProperty._observers.length
0 // ok
> heightProperty._observers.length
0 // ok
> areaProperty.dependencies.length
2 // should be 0
> areaProperty.dependencyListeners.length
2 // should be 0
```

I'd like one other developer to confirm that I'm interpreting this correctly. Assigning to @samreid since he was the primary author of DerivedProperty.
